osu.Game.Rulesets.Catch/UI/PoolableHitExplosion.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Pooling;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Skinning;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.UI
+{
+    public class PoolableHitExplosion : PoolableDrawable
+    {
+        public const double DURATION = 1000;
+
+        private Color4 objectColour;
+
+        public Color4 ObjectColour
+        {
+            get => objectColour;
+            set
+            {
+                if (objectColour == value)
+                    return;
+
+                objectColour = value;
+
+                if (skinnableExplosion.Drawable is ICatchHitExplosion hitExplosion)
+                {
+                    hitExplosion.ObjectColour = value;
+                }
+            }
+        }
+
+        public PalpableCatchHitObject HitObject;
+
+        public JudgementResult JudgementResult;
+
+        public float CatcherWidth;
+
+        public float CatchPosition;
```

Comment:
We generally prefer auto-properties to plain fields. Although it's not a super hard rule.